src/transformers/cache_utils.py

```diff
@@ -688,6 +688,10 @@ def get_seq_length(self, layer_idx: Optional[int] = 0) -> int:
 
     def get_max_length(self) -> Optional[int]:
         """Returns the maximum sequence length of the cached states."""
+        logger.warning_once(
+            "`get_max_length` for SinkCache is deprecated and will return 0 after v4.47. ",
+            "Use past_key_values.window_length to get the maximum window length of the cache.",
+        )
```

Comment:
I dislike this change -- `self.window_length` is the maximum length of the cache, returning 0 (or `None`) would not be correct. I realize the use of the method is inconsistent throughout our cache classes.

We should add two new attributes to the `Cache` classes, to fully specify a cache:
- a boolean attribute to determine whether the cache grows or has a static shape (discussed in other PRs, to be used in compilation checks)
- a boolean to determine whether it uses a sliding window (to be used e.g. in this PR)

With `get_max_length` and these two attributes we could be able to define all types of cache, and control the modeling code accordingly 🤗

Reply:
Hmm, makes sense. I used the `none` following SlidingwindownCache which actually is almost same as SinkCache but does not return any max length. So we need two attributes on each cache class: `_is_sliding` and `_is_static` right? What about HybridCache, will it be sliding or not?

Reply:
Those two attribute names look good 👍 (btw, probably should be done in a separate PR, for better future reference)

`HybridCache` is a good corner case indeed 🤔 I think we can treat it as a NON sliding cache, as common model variables (e.g. `attention_mask`) need to be prepared according to the non-sliding cache layers.

Reply:
Got you! I will add those two attributes in a second PR. 

Just so we're on the same page, even after adding attributes we'll have to crop inputs to the max sliding window length. I guess the attr will make checks look nicer, like `if cache._is_static: length = cache.get_max_length()`

Reply:
Yes, on the same page as you: we need to crop things with sliding windows :)